File: lab4 mpp/TeledonPersistence/DonatorDBRepository.cs
using teledon.domain;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace teledon.repository
{
    public class DonatorDBRepository : IDonatorRepo
    {

        private static readonly ILog log = LogManager.GetLogger("RepoDonator");

        public Donator Delete(long id)
        {
            log.InfoFormat("entering delete with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Donatori where idDonator=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();

            }
            log.InfoFormat("exiting delete");
            return null;
        }

        public IEnumerable<Donator> FindAll()
        {
            log.InfoFormat("Entering findAll");
            IDbConnection con = DBUtils.getConnection();
            IList<Donator> donatori = new List<Donator>();
            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Donatori";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        long id = dataR.GetInt32(0);
                        string AdresaDonator = dataR.GetString(1);
                        string Telefon = dataR.GetString(2);
                        string Nume = dataR.GetString(3);
                        string Prenume = dataR.GetString(4);
                        Donator donator = new Donator(AdresaDonator, Telefon, Nume, Prenume);
                        donator.ID = id;
                        donatori.Add(donator);
                    }
                }
            }
            log.InfoFormat("Exiting findAll with the result from it");
            return donatori; 
        }

        public Donator FindOne(long id)
        {

            log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection();
            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Donatori where idDonator=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idDonator = dataR.GetInt32(0);
                        string Adresa = dataR.GetString(1);
                        string Telefon = dataR.GetString(2);
                        string Nume = dataR.GetString(3);
                        string Prenume = dataR.GetString(4);

                        //   long id = long.Parse(dataR.GetString(3));

                        Donator donator = new Donator(Adresa, Telefon, Nume, Prenume);
                        donator.ID = idDonator;
                        log.InfoFormat("Entering findOne with value {0}", idDonator);
                        return donator;
                    }
                }
            }
            log.InfoFormat("Entering findOne with value {0}", null);
            return null;
        }

        public Donator Save(Donator entity)
        {
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "Insert into Donatori(adresa, telefon, nume, prenume)  values (@adresa, @telefon, @nume, @prenume)";
                
                IDbDataParameter paramAdresa = comm.CreateParameter();
                paramAdresa.ParameterName = "@adresa";
                paramAdresa.Value = entity.Adresa;
                comm.Parameters.Add(paramAdresa);

                IDbDataParameter paramTelefon = comm.CreateParameter();
                paramTelefon.ParameterName = "@telefon";
                paramTelefon.Value = entity.Telefon;
                comm.Parameters.Add(paramTelefon);

                IDbDataParameter paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@nume";
                paramNume.Value = entity.Nume;
                comm.Parameters.Add(paramNume);

                IDbDataParameter paramPrenume = comm.CreateParameter();
                paramPrenume.ParameterName = "@prenume";
                paramPrenume.Value = entity.Prenume;
                comm.Parameters.Add(paramPrenume);

                var result = comm.ExecuteNonQuery();
              
                if (result == 0)
                    log.InfoFormat("save exited with value {0}", result);
                else
                    log.InfoFormat("save exited with value {0}", result);

            }


            return null;
        }

        public Donator FindByNume(string nume, string prenume)
        {
            log.InfoFormat("Entering findByName with value {0}, {1}", nume, prenume);
            IDbConnection con = DBUtils.getConnection();
            Console.WriteLine(con);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select * from Donatori where nume=@nume and prenume=@prenume";
                IDbDataParameter paramNume = comm.CreateParameter();
                paramNume.ParameterName = "@nume";
                paramNume.Value = nume;
                comm.Parameters.Add(paramNume);
                IDbDataParameter paramPrenume = comm.CreateParameter();
                paramPrenume.ParameterName = "@prenume";
                paramPrenume.Value = prenume;
                comm.Parameters.Add(paramPrenume);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        long idDonator = dataR.GetInt32(0);
                        string Adresa = dataR.GetString(1);
                        string Telefon = dataR.GetString(2);
                        string Nume = dataR.GetString(3);
                        string Prenume = dataR.GetString(4);

                        //   long id = long.Parse(dataR.GetString(3));

                        Donator donator = new Donator(Adresa, Telefon, Nume, Prenume);
                        donator.ID = idDonator;
                        if (donator.ID == null)
                            return null;
                        log.InfoFormat("Exiting findByNume with value {0}", idDonator);
                        return donator;
                    }
                }
            }
            log.InfoFormat("Entering findOne with value {0}", null);
            return null;
        }

        public Donator Update(Donator entity)
        {
            throw new NotImplementedException();
        }
    }
}

File: GUImpp/TeledonPersistence/src/main/java/teledon/persistence/repository/DonatorDBRepository.java
package teledon.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import teledon.model.Donator;

import java.sql.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class DonatorDBRepository implements IDonatorRepo {

    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();

    public DonatorDBRepository(Properties properties) {
        logger.info("Initializing DonatorDBRepository with properties: {} ", properties);
        dbUtils = new JdbcUtils(properties);
    }

    @Override
    public Donator findByNume(String nume, String prenume) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        Donator donator = new Donator();
        try(PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Donatori WHERE nume=? AND prenume=?")) {
            preparedStatement.setString(1, nume);
            preparedStatement.setString(2, prenume);
            ResultSet result = preparedStatement.executeQuery();
            Long id = result.getLong("idDonator");
            String adresa = result.getString("adresa");
            String telefon = result.getString("telefon");
            String n = result.getString("nume");
            String p = result.getString("prenume");
            donator = new Donator(adresa, telefon, nume, prenume);
            donator.setId(id);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " +e);
        }
        logger.traceExit(donator);
        if(donator.getId() == null)
            return null;
        return donator;
    }

    @Override
    public Donator save(Donator entity) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into Donatori(adresa, telefon, nume, prenume) values (?,?,?,?)")){
            preparedStatement.setString(1, entity.getAdresa());
            preparedStatement.setString(2, entity.getTelefon());
            preparedStatement.setString(3, entity.getNume());
            preparedStatement.setString(4, entity.getPrenume());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }
        catch(SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return entity;

    }

    @Override
    public Donator update(Donator entity) {
        return null;
    }

    @Override
    public Donator findOne(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        Donator donator = new Donator();
        try(PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Donatori WHERE idDonator=?")) {
                preparedStatement.setLong(1, aLong);
                ResultSet result = preparedStatement.executeQuery();
                String adresa = result.getString("adresa");
                String telefon = result.getString("telefon");
                String nume = result.getString("nume");
                String prenume = result.getString("prenume");
                donator = new Donator(adresa, telefon, nume, prenume);
                donator.setId(aLong);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " +e);
        }
        logger.traceExit(donator);
        return donator;
    }

    @Override
    public Donator delete(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("DELETE FROM Donatori WHERE idDonator = ?")){
            preparedStatement.setInt(1, aLong.intValue());
            int result = preparedStatement.executeUpdate();
            logger.trace("Deleted {} instances", result);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        return null;
    }

    @Override
    public Iterable<Donator> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Donator> donatori = new ArrayList<>();

        try (PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Donatori")) {
            try(ResultSet result = preparedStatement.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("idDonator");
                    String adresa = result.getString("adresa");
                    String telefon = result.getString("telefon");
                    String nume = result.getString("nume");
                    String prenume = result.getString("prenume");
                    Donator donator = new Donator(adresa, telefon, nume, prenume);
                    donator.setId(id);
                    donatori.add(donator);
                }
            }
        } catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(donatori);
        return donatori;
    }
}
